From a1de5eb3e2641887f62cfdec89e238ed76fd41a0 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Sat, 25 Jan 2025 17:19:02 +0800
Subject: [PATCH 2/7] FEDORA: immodule: fix a GCC warning

Signed-off-by: Cosimo Cecchi <cosimoc@gnome.org>

[Mingcong Bai: This is now an error:

gtk/gtkimmodule.c:665:49: error: passing argument 3 of 'g_hash_table_lookup_extended'
 from incompatible pointer type [-Wincompatible-pointer-types]
  665 |                                                 &context_id, NULL);
      |                                                 ^~~~~~~~~~~
      |                                                 |
      |                                                 gchar ** {aka char **}

Possibly due to recent GLib API changes.]

Link: https://src.fedoraproject.org/rpms/gtk2/blob/a3a398f4f95684dba7a461acf389626eb2a3b90e/f/gtk2-c89-2.patch
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 gtk/gtkimmodule.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gtk/gtkimmodule.c b/gtk/gtkimmodule.c
index 1fcb17a7d8..445b2be87f 100644
--- a/gtk/gtkimmodule.c
+++ b/gtk/gtkimmodule.c
@@ -662,7 +662,7 @@ lookup_immodule (gchar **immodules_list)
 	  gboolean found;
 	  gchar *context_id;
 	  found = g_hash_table_lookup_extended (contexts_hash, *immodules_list,
-						&context_id, NULL);
+						(gpointer *) &context_id, NULL);
 	  if (found)
 	    return context_id;
 	}
-- 
2.48.1

